src/components/GestureButtons.tsx

```diff
@@ -1,65 +1,108 @@
+}
 
-/* Buttons */
+export interface BaseButtonProperties extends RawButtonProperties {
+  onPress?: (pointerInside: boolean) => void;
+  onActiveStateChange?: (active: boolean) => void;
+  style?: StyleProp<ViewStyle>;
+}
 
-export class BaseButton extends React.Component {
-  static propTypes = {
-    ...RawButton.propTypes,
-    onPress: PropTypes.func,
-    onActiveStateChange: PropTypes.func,
-  };
+export interface RectButtonProperties extends BaseButtonProperties {
+  underlayColor?: string;
+  activeOpacity?: number;
+}
+
+export interface BorderlessButtonProperties extends BaseButtonProperties {
+  borderless?: boolean;
+  activeOpacity?: number;
+}
 
-  constructor(props) {
+export const RawButton: React.ComponentType<RawButtonProperties> = createNativeWrapper(
+  GestureHandlerButton,
+  {
+    shouldCancelWhenOutside: false,
+    shouldActivateOnStart: false,
+  }
+);
+
+export class BaseButton extends React.Component<BaseButtonProperties> {
+  constructor(props: BaseButtonProperties) {
     super(props);
-    this._lastActive = false;
+    this.lastActive = false;
   }
 
-  _handleEvent = ({ nativeEvent }) => {
+  private lastActive: boolean;
+
+  private handleEvent = ({
+    nativeEvent,
+  }: HandlerStateChangeEvent<NativeViewGestureHandlerPayload>) => {
     const { state, oldState, pointerInside } = nativeEvent;
     const active = pointerInside && state === State.ACTIVE;
 
-    if (active !== this._lastActive && this.props.onActiveStateChange) {
+    if (active !== this.lastActive && this.props.onActiveStateChange) {
       this.props.onActiveStateChange(active);
     }
 
     if (
       oldState === State.ACTIVE &&
       state !== State.CANCELLED &&
-      this._lastActive &&
+      this.lastActive &&
       this.props.onPress
     ) {
       this.props.onPress(active);
     }
 
-    this._lastActive = active;
+    this.lastActive = active;
   };
 
   // Normally, the parent would execute it's handler first,
   // then forward the event to listeners. However, here our handler
   // is virtually only forwarding events to listeners, so we reverse the order
   // to keep the proper order of the callbacks (from "raw" ones to "processed").
-  _onHandlerStateChange = (e) => {
-    this.props.onHandlerStateChange && this.props.onHandlerStateChange(e);
-    this._handleEvent(e);
+  private onHandlerStateChange = (
+    e: HandlerStateChangeEvent<NativeViewGestureHandlerPayload>
+  ) => {
+    this.props.onHandlerStateChange?.(e);
+    this.handleEvent(e);
   };
 
-  _onGestureEvent = (e) => {
-    this.props.onGestureEvent && this.props.onGestureEvent(e);
-    this._handleEvent(e);
+  private onGestureEvent = (
+    e: GestureEvent<NativeViewGestureHandlerPayload>
+  ) => {
+    this.props.onGestureEvent?.(e);
+    this.handleEvent(
+      e as HandlerStateChangeEvent<NativeViewGestureHandlerPayload>
+    ); // TODO: maybe it is not correct
```

Comment:
Type cast + TODO.

Reply:
`this.handleEvent` is called both in `onGestureEvent` and `onHandlerStateChange`, which take different types of native events, so one of the calls must cast the type unfortunately. Maybe it is a mistake in implementation. `handleEvent` uses `oldState`, which is available only in `HandlerStateChangeEvent<NativeViewGestureHandlerPayload>`, so I cast the former into it. From testing it in example, `oldState` is never present in `onGestureEvent`'s native event. It would be good to check how to correctly resolve this.